src/main/java/com/fasterxml/jackson/databind/util/TokenBuffer.java

```diff
@@ -442,6 +449,8 @@ public void serialize(JsonGenerator gen) throws IOException
                         gen.writeNumber((Long) n);
                     } else if (n instanceof Short) {
                         gen.writeNumber((Short) n);
+                    } else if (n instanceof LazyNumber) {
+                        gen.writeNumber(((LazyNumber) n).getText());
```

Comment:
Potential problem here would be that binary formats do not support "Number as Strings".
But there is method `JsonGenerator.canWriteFormattedNumbers()` that actually allows checking if generator implements support for `writeNumber(String)` or not (word "formatted" is bit misnomer here).

Reply:
I have now added a new local method that checks `canWriteFormattedNumbers`